### PR TITLE
Remove hyperlinks for obsolete PersistentVolume types

### LINKED ISSUE #42868: Unknown References in Types of Persistent Volumes
**This is a Bug Report**

**Problem:** In the [Types of Persistent Volumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#types-of-persistent-volumes) section of the [Persistent Volumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) page, there are some `PersistentVolume` types that were supported by an older version of `Kubernetes.` Those `PersistentVolume` types are not present in the documentation as they have been removed.

**Proposed Solution:** To remove the references from the hyperlinks from the `PersistentVolume` can be a solution, as there is no information present in the recent release of `Kuberetes.`

**Page to Update:**
https://kubernetes.io/docs/concepts/storage/persistent-volumes/

### PR DESCRIPTION
**Pull request by thisisharrsh**
This PR fixes the references of [`Types of Persistent Volumes`](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#types-of-persistent-volumes) section in [`Persistent Volumes`](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) page. 

Fixes #42868

### CODE REVIEW

**Comment by thisisharrsh** [comment]
Hello @msau42 @xing-yang, please review my PR.

File: content/en/docs/concepts/storage/persistent-volumes.md [hunk 1]
```diff
@@ -541,13 +541,13 @@ Older versions of Kubernetes also supported the following in-tree PersistentVolu
...
   (**not available** starting v1.21)
-* [`flocker`](/docs/concepts/storage/volumes/#flocker) - Flocker storage
+* `flocker` - Flocker storage.
   (**not available** starting v1.25)
-* [`quobyte`](/docs/concepts/storage/volumes/#quobyte) - Quobyte volume
+* `quobyte` - Quobyte volume.
   (**not available** starting v1.25)
-* [`storageos`](/docs/concepts/storage/volumes/#storageos) - StorageOS volume
+* `storageos` - StorageOS volume.
   (**not available** starting v1.25)
```
Comment (aj11anuj) [inline content/en/docs/concepts/storage/persistent-volumes.md:551]: 1. These 4 icons will now look like `photonPersistentDisk`.

2. I think we should also try making these 5 icons (`photonPersistentDisk`, `scaleIO`, `flocker`, `quobyte`, `storageos`) bold as well inside there code block as after removing hyperlink there text is looking very thin and light. Have a look at `photonPersistentIcon` at this page on K8s site.

**Review by aj11anuj — COMMENTED** [review]
Here are some of my thoughts, what do you think?

**Comment by dipesh-rawat** [comment]
/retitle Remove hyperlinks for obsolete PersistentVolume types

File: content/en/docs/concepts/storage/ephemeral-volumes.md
Comment (dipesh-rawat) [inline content/en/docs/concepts/storage/ephemeral-volumes.md:1]: The changes in this file appears unrelated to the issue (https://github.com/kubernetes/website/issues/42868) the PR aims to address. Could you kindly consider removing it from this PR?
